riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyRequestSizeValidationComponentTest.java

```diff
@@ -0,0 +1,236 @@
+package com.nike.riposte.server.componenttest;
+
+import com.nike.riposte.server.http.*;
+import io.restassured.response.ExtractableResponse;
+import com.nike.riposte.server.Server;
+import com.nike.riposte.server.config.ServerConfig;
+import com.nike.riposte.server.testutils.ComponentTestUtils;
+import com.nike.riposte.util.Matcher;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class VerifyRequestSizeValidationComponentTest {
+
+
+    private static Server server;
+    private static ServerConfig serverConfig;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        serverConfig = new RequestSizeValidationConfig();
+        server = new Server(serverConfig);
+        server.startup();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.shutdown();
+    }
+
+    @Test
+    public void should_return_bad_request_when_request_exceeds_global_configured_max_request_size() {
+        ExtractableResponse response =
+                given()
+                        .baseUri("http://127.0.0.1")
+                        .port(serverConfig.endpointsPort())
+                        .basePath(BasicEndpoint.MATCHING_PATH)
+                        .log().all()
+                        .body("{'key1':'value1'}")
```

Comment:
I'd suggest that the global and endpoint-override max request size values be constants that can be referenced by these tests, and dynamically generate a request of the appropriate size in bytes relative to whatever constant you're testing against.

e.g. for this test:
```
.body(generatePayloadOfSizeInBytes(GLOBAL_MAX_REQUEST_SIZE + 1))
```